x-pack/test/apm_api_integration/basic/tests/service_overview/transaction_groups.ts

```diff
@@ -0,0 +1,235 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import expect from '@kbn/expect';
+import qs from 'querystring';
+import { pick, uniqBy } from 'lodash';
+import { expectSnapshot } from '../../../common/match_snapshot';
+import { FtrProviderContext } from '../../../../common/ftr_provider_context';
+import archives from '../../../common/archives_metadata';
+
+export default function ApiTest({ getService }: FtrProviderContext) {
+  const supertest = getService('supertest');
+  const esArchiver = getService('esArchiver');
+
+  const archiveName = 'apm_8.0.0';
+  const { start, end } = archives[archiveName];
+
+  describe('Service overview transaction groups', () => {
+    describe('when data is not loaded', () => {
+      it('handles the empty state', async () => {
+        const response = await supertest.get(
+          `/api/apm/services/opbeans-java/overview_transaction_groups?${qs.stringify({
+            start,
+            end,
+            uiFilters: '{}',
+            size: 5,
+            numBuckets: 20,
+            pageIndex: 0,
+            sortDirection: 'desc',
+            sortField: 'impact',
+          })}`
+        );
```

Comment:
I've started using the `url.format` method which is little cleaner. Nothing crucial.

```ts
import { format } from 'url';
const url = format({
  pathname: `/api/apm/services/opbeans-java/overview_transaction_groups`,
  query: {
    start,
    end,
    uiFilters: '{}',
    size: 5,
    // ...
  },
});
```